Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml

```diff
@@ -16731,4 +16735,14 @@
         <EditorCategories value="Race:Zerg"/>
     </CUpgrade>
     <CUpgrade id="AP_SILiberatorDefenderMode"/>
+    <CUpgrade id="AP_AberrationBanelingLaunch">
+        <EffectArray Operation="Set" Reference="Abil,AP_BanelingLaunchHangar,InfoArray[Ammo1].Button.Requirements" Value="AP_BanelingLaunchArm"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_BanelingLaunchHangar,MaxCount" Value="1"/>
+    </CUpgrade>
+    <CUpgrade id="AP_AberrationBanelingLaunchLevel2">
+        <EffectArray Operation="Set" Reference="Effect,AP_BanelingLaunchImpact,EffectArray[0]" Value="AP_BanelingLaunchSet"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_BanelingLaunchHangar,MaxCount" Value="3"/>
+        <EffectArray Operation="Set" Reference="Button,AP_BanelingLaunch,Tooltip" Value="Button/Tooltip/AP_BanelingLaunchLevel2"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_BanelingLaunchHangar,InfoArray[Ammo1].Button.Requirements" Value="AP_BanelingLaunchArmLevel2"/>

```

Comment:
Are requirements an upgradeable field? We generally don't use this style of requirement changing, instead having the requirement node tree check if the player has the upgrade. This might still work, but it would need eyes on it in testing.

Reply:
From my tests, it works, and made the requirement node tree a lot less painful to navigate, as I don't need multiple nested node comparisons.

It is used for example for the reaver magazine to change it from stopping at queue 10 instead of queue 5 after the upgrade.